data/abilities.js

```diff
@@ -579,9 +579,14 @@ exports.BattleAbilities = {
 		onStart: function (pokemon) {
 			this.add('-ability', pokemon, 'Dark Aura');
 		},
-		onAnyBasePower: function (basePower, source, target, move) {
+		onAnyTryPrimaryHit: function (target, source, move) {
 			if (target === source || move.category === 'Status' || move.type !== 'Dark') return;
-			return this.chainModify([move.hasAuraBreak ? 0x0C00 : 0x1547, 0x1000]);
+			move.auraBoost = 0x1547;
+		},
+		onAnyBasePower: function (basePower, source, target, move) {
+			if (!move.auraBoost) return;
+			this.chainModify([move.auraBoost, 0x1000]);
+			delete move.auraBoost;
```

Comment:
There's no reason to delete it, is there?

Reply:
You can just... set it, and then just return if it's truthy?

Reply:
Aura is a tricky ability, as it only gains one boost no matter how many Auras are on the field, and you don't have to have the Aura yourself. This means that you need some way of making sure only one Aura actually ends up applying the boost.

Reply:
> You can just... set it, and then just return if it's truthy?

That would be the way of making sure only one aura actually ends up applying the boost.

Reply:
I don't understand your proposal either, @Zarel 

We already return if falsey...

Reply:
2923b9414021615214 is my proposal.

Reply:
For the record, replacing `delete move.auraBoost` with `move.auraBoost = null` would at least be an improvement for V8 perf, but my approach also makes it possible for other effects to detect whether the move has been aura-boosted, and is fewer lines and is easier to read (at least to me).

Reply:
Should `CONTRIBUTING.md` mention constructs to avoid such as `delete`?

Reply:
Yeah.